providers/fips/fipsprov.c

```diff
@@ -1238,3 +1259,128 @@ void OSSL_INDICATOR_get_callback(OSSL_LIB_CTX *libctx,
             *cb = NULL;
     }
 }
+
+/* On Deamnd test infrastructure */
+
+struct ondemand_test_status {
+    const char *alg;
+    int pass;
+};
+
+static struct ondemand_test_status ondemand_kat_integrity_tests[] = {
+    { NULL, 0 }
+};
+static struct ondemand_test_status ondemand_kat_cipher_tests[] = {
+    { NULL, 0 }
+};
+static struct ondemand_test_status ondemand_kat_asym_cipher_tests[] = {
+    { NULL, 0 }
+};
+static struct ondemand_test_status ondemand_kat_asym_keygen_tests[] = {
+    { "SLH-DSA-SHA2-128f", 0 },
+    { NULL, 0 }
+};
+static struct ondemand_test_status ondemand_kat_kem_tests[] = {
+    { NULL, 0 }
+};
+static struct ondemand_test_status ondemand_kat_digest_tests[] = {
+    { NULL, 0 }
+};
+static struct ondemand_test_status ondemand_kat_signature_tests[] = {
+    { "SLH-DSA-SHAKE-128f", 0 },
+    { NULL, 0 }
+};
+static struct ondemand_test_status ondemand_kat_kdf_tests[] = {
+    { NULL, 0 }
+};
+static struct ondemand_test_status ondemand_kat_ka_tests[] = {
+    { NULL, 0 }
+};
+
+static struct ondemand_test_status *ondemand_tests_status[OSSL_ONDEMAND_MAX] = {
+    ondemand_kat_integrity_tests,
+    ondemand_kat_cipher_tests,
+    ondemand_kat_asym_cipher_tests,
+    ondemand_kat_asym_keygen_tests,
+    ondemand_kat_kem_tests,
+    ondemand_kat_digest_tests,
+    ondemand_kat_signature_tests,
+    ondemand_kat_kdf_tests,
+    ondemand_kat_ka_tests,
+};
+
+int FIPS_kat_on_demand(OSSL_LIB_CTX *libctx, int type, const char *alg)
+{
+    FIPS_GLOBAL *fgbl;
+    OSSL_SELF_TEST *ev = NULL;
+    void *local_key;
+    int idx = 0;
+    int ret = 0;
+
+    if ((fgbl = ossl_lib_ctx_get_data(libctx, OSSL_LIB_CTX_FIPS_PROV_INDEX)) == NULL)
```

Comment:
Note that this is an expensive operation.  It is not a problem given them way SLH-DSA uses a global flag but it could become one if other algorithms get naïvely converted.